solutions/downloader/DownloadThread.java

```
@@ -0,0 +1,91 @@
+// Worker thread responsible for downloading a part of the file.
+package downloader;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedReader;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.RandomAccessFile;
+import java.net.Socket;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.util.logging.Logger;
+
+public class DownloadThread extends Thread {
+	private final static Logger logger = Logger.getLogger(FileDownloader.class.getName());	
+	private final static String newLine = "\r\n";
+	private static final int BUFFER_SIZE = 4096;
+	private int startByte;
+	private int endByte;
+	private Socket socket;
+	private BufferedInputStream inFromServer = null;
+	private DataOutputStream outToServer = null;
+	private RandomAccessFile outputFile;	
+	
+	public DownloadThread(int startByte, int endByte) {		
+		this.startByte = startByte;
+		this.endByte = endByte;
+		
+		try {
+			outputFile = new RandomAccessFile(FileDownloader.fileName, "rw");
+		} catch (FileNotFoundException e) {
+			logger.warning(e.getMessage());
```

Reraise an exception? This silent failure is dangerous. Sure you can figure it out through logging, but if 1 of the 100 threads waits, there's no easy way for your application to detect and handle it.